widget/popup.go

```diff
@@ -181,17 +179,20 @@ func (r *popUpRenderer) Destroy() {
 }
 
 type modalPopUpRenderer struct {
-	center  fyne.Layout
 	popUp   *PopUp
 	bg      *canvas.Rectangle
 	objects []fyne.CanvasObject
 }
 
-func (r *modalPopUpRenderer) Layout(size fyne.Size) {
-	r.center.Layout(r.objects, size)
+func (r *modalPopUpRenderer) Layout(canvasSize fyne.Size) {
+	size := r.popUp.innerSize.Subtract(fyne.NewSize(theme.Padding()*2, theme.Padding()*2))
+	size = size.Min(canvasSize).Max(r.popUp.MinSize())
```

Comment:
This one is hard to understand.
We want the content's size to be at least `r.popUp.MinSize`.
If `size` is bigger, we want it to be `size`.
But if `size` is bigger than the canvas, we want the content's size to be the canvas' size.
Maybe splitting the line and change the order helps the reader to understand what happens here?

```suggestion
	requestedSize := r.popUp.innerSize.Subtract(fyne.NewSize(theme.Padding()*2, theme.Padding()*2))
	size = r.popUp.MinSize().Max(requestedSize)
	size = size.Min(canvasSize)
```

Please not how I changed the order of the operations to the order of the expectations I wrote above.

N.B.: There are no tests for these computations. Please add some.